base_control/Plot.cpp

```diff
@@ -0,0 +1,55 @@
+#include "Plot.h"
+//////////////////////////////////Constructor//////////////////////////////////////
+Plot::Plot(Movement *moveAll) {
+    moveAll_ = moveAll;
+    time_msg_ = millis();
+}
+
+//////////////////////////////////Plot Functions//////////////////////////////////////
+void Plot::PlotMotorSpeed() {
+    PlotData(
+        moveAll_->back_left_motor_.getCurrentSpeed(),
+        moveAll_->front_left_motor_.getCurrentSpeed(),
+        moveAll_->front_right_motor_.getCurrentSpeed(),
+        moveAll_->back_right_motor_.getCurrentSpeed()
+    );
+}
+
+void Plot::PlotMotorTicks() {
+    PlotData(
+        moveAll_->back_left_motor_.getLastTicks(),
+        moveAll_->front_left_motor_.getLastTicks(),
+        moveAll_->front_right_motor_.getLastTicks(),
+        moveAll_->back_right_motor_.getLastTicks()
+    );
+}
+
+void Plot::PlotTargetandCurrent() {
+    PlotData(
+        moveAll_->back_right_motor_.getTargetRpm(moveAll_->getTargetLinearVelocity()),
+        moveAll_->back_right_motor_.getCurrentSpeed(),
+        moveAll_->front_right_motor_.getTargetRpm(moveAll_->getTargetLinearVelocity()),
+        moveAll_->front_right_motor_.getCurrentSpeed()
+    );
+}
+
+void Plot::PlotData(const double data1, const double data2, const double data3, const double data4) {
+     if(millis()-time_msg_>35) {
+        return;
+    }
+
+    const byte* byteData1 = (byte*)(&data1);

```

Comment:
can you do a static_cast<byte*>?

Reply:
No, it is marked as an invalid static_cast. According to what i read. There exists not subtype relationship between these two types.